### PR TITLE
redlib: init at 0.31.0

### LINKED ISSUE #292634: Package request: redlib
**Project description**

> Private front-end for Reddit. 

This is continuation of the Libreddit project, which is already packaged in nixpkgs. Thus, the Libreddit module could probably be reused, if Redlib was packaged. 
> While originating as a fork of Libreddit, the name "Redlib" was adopted to avoid legal issues, as Reddit only allows the use of their name if structured as "XYZ For Reddit". 

**Mtadata**

* source URL: https://github.com/redlib-org/redlib
* license:  AGPL-3.0
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by soispha**
## Description of changes

This adds [redlib](https://github.com/redlib-org/redlib) and closes #292634. It only includes the package, as the libreddit module can be reused.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12564,6 +12564,10 @@ with pkgs;
 
   redir = callPackage ../tools/networking/redir { };
 
+  redlib = callPackage ../by-name/re/redlib/package.nix {
+    inherit (darwin.apple_sdk.frameworks) Security;
```
Comment (Aleksanaa) [inline pkgs/top-level/all-packages.nix:12568]: Why do we need this inherit?

File: pkgs/by-name/re/redlib/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  env = {
+    SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+  };
+
+  passthru.tests = {
+    inherit (nixosTests) redlib;
+  };
+
+  meta = {
+    description = "Private front-end for Reddit ";
```
Comment (Aleksanaa) [inline pkgs/by-name/re/redlib/package.nix:55]:
```suggestion
    description = "Private front-end for Reddit";
```

Also mention it's a fork of libreddit?

File: pkgs/by-name/re/redlib/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, cacert
+, nixosTests
+, rustPlatform
+, fetchFromGitHub
+, Security
+,
```
Comment (Aleksanaa) [inline pkgs/by-name/re/redlib/package.nix:8]:
```suggestion
```

File: nixos/tests/redlib.nix
Comment (Aleksanaa) [inline nixos/tests/redlib.nix:1]: Please add this test in a separate commit.

**Comment by soispha** [comment]
Result of `nixpkgs-review pr 292645` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>redlib</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/top-level/all-packages.nix:12568]: But `darwin` is in its args. I assume in `by-name` we should avoid using arbitrary callPackage function when possible.

Comment (soispha) [reply pkgs/top-level/all-packages.nix]: Thank you, I didn't know about that.

**Comment by Aleksanaa** [comment]
Please amend (squash?) the fixup commit into original commit. Also another commit should be "nixos/tests: add redlib test using libreddit module", as the redlib module doesn't exist yet

**Comment by soispha** [comment]
I was going to run `git rebase --autosquash` to automatically rebase the fixup commits when the pr is approved, but I can obviously do this for every push

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/re/redlib/package.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  env = {
+    SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+  };
+
+  passthru.tests = {
+    inherit (nixosTests) redlib;
+  };
+
+  meta = {
```
Comment (drupol) [inline pkgs/by-name/re/redlib/package.nix:53]: Missing `meta.mainProgram`

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/re/redlib/package.nix (hunk 4 above)
Comment (drupol) [reply pkgs/by-name/re/redlib/package.nix:53]: Please sort `meta` attributes too.

Comment (soispha) [reply pkgs/by-name/re/redlib/package.nix:53]: I've added it

**Comment by drupol** [comment]
Don't forget to rebase ;)

**Comment by soispha** [comment]
Okay, rebased and squashed all fixup commits.

**Review by drupol — APPROVED** [review]

**Comment by soispha** [comment]
The NixOS test in this pr somehow failed to test, if changing the package for the `libreddit` module even works (it does not). I've opened a follow up pr that changes the `libreddit` services so it does: #300463.

**Comment by trofi** [comment]
Looks like nixos test was not added to the test list. That caused `redlib.tests` to fail to evaluate. Proposed the change as:
- https://github.com/NixOS/nixpkgs/pull/305925

**Pull request merged**